src/components/i18nBanner.vue

```diff
@@ -0,0 +1,139 @@
+<!--
+  This source file is part of the Swift.org open source project
+
+  Copyright (c) 2023 Apple Inc. and the Swift project authors
+  Licensed under Apache License v2.0 with Runtime Library Exception
+
+  See https://swift.org/LICENSE.txt for license information
+  See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+-->
+
+<template>
+<div v-if="displayi18nBanner" class="i18n-banner">
+  <div class="i18n-banner__wrapper">
+    <router-link
+      :to="link"
+      @click.native="setPreferredLocale(preferredLocale)"
+      class="i18n-banner__link"
+      :lang="getCodeForSlug(preferredLocale)"
+    >{{ $i18n.messages[preferredLocale]['view-in'] }}<InlineChevronRightIcon class="icon-inline" />
+    </router-link>
+    <div class="i18n-banner__close-icon-wrapper">
+      <button
+        class="i18n-banner__close-icon-button"
+        :aria-label="$t('continue-viewing')"
+        @click="setPreferredLocale($i18n.locale)"
+      >
+        <CloseIcon class="icon-inline" />
+      </button>
+    </div>
+  </div>
+</div>
+</template>
+<script>
+import InlineChevronRightIcon from 'theme/components/Icons/InlineChevronRightIcon.vue';
+import CloseIcon from 'theme/components/Icons/CloseIcon.vue';
+import locales from 'theme/lang/locales.json';
+import { defaultLocale } from 'theme/lang/index.js';
+import AppStore from 'docc-render/stores/AppStore';
+import { getCodeForSlug } from 'docc-render/utils/i18n-utils';
+
+export default {
+  name: 'i18nBanner',
+  components: {
+    InlineChevronRightIcon,
+    CloseIcon,
+  },
+  computed: {
+    preferredLocale: () => {
+      const appPreferredLocale = AppStore.state.preferredLocale;
+      // if user has a storaged preferred locale, return it
+      if (appPreferredLocale) return appPreferredLocale;
+      // find if user's navigator preference is available with the locales we provide
+      // in case it is, that will be the preferred locale to display
+      return locales.find((locale) => {
+        const languageAvailable = locale.code.split('-')[0];
+        const navigatorLanguage = window.navigator.language;
+        return navigatorLanguage.includes(languageAvailable);
```

Comment:
I wonder if `includes` could return a false positive... we are comparing `en` to `en` in most cases, right?

Reply:
true, I'm comparing the languages now: https://github.com/apple/swift-docc-render/pull/554/commits/20363e62b74e680959ce486e7e1ce24a3ee2469c